### PR TITLE
feat!: upgrade to Angular 15

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [x] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

### CODE REVIEW

File: libs/view-transition/src/services/view-transition.service.ts [hunk 1]
```diff
@@ -6,10 +6,12 @@ import {Observable, throwError} from 'rxjs';
     providedIn: 'root',
 })
 export class ViewTransitionService {
-    private readonly isSupported = 'startViewTransition' in this.document;
-
     constructor(@Inject(DOCUMENT) private readonly document: Document) {}
 
+    get isSupported(): boolean {
+        return 'startViewTransition' in this.document;
```
Comment (nsbarsukov) [inline libs/view-transition/src/services/view-transition.service.ts:12]: Why it should be getter ?

Comment (splincode) [reply libs/view-transition/src/services/view-transition.service.ts:12]: image

File: libs/payment-request/tests/payment-request.service.spec.ts [hunk 2]
```diff
@@ -44,20 +40,16 @@ describe('Payment Request Service', () => {
...
-        } as PaymentResponse;
+        } as Partial<PaymentResponse>;
 
-        service = TestBed.get(PaymentRequestService);
+        service = TestBed.inject(PaymentRequestService);
     });
 
     it('throws an error if Payment Request in cannot pay status', done => {
-        (global as any).PaymentRequest = FakePaymentRequest;
+        (window as any).PaymentRequest = FakePaymentRequest;
```
[the remark this answers is not in the document]
Comment (splincode) [reply libs/payment-request/tests/payment-request.service.spec.ts]: fixed

File: libs/payment-request/src/directives/payment-submit/payment-submit.directive.ts [hunk 3]
```diff
@@ -22,11 +21,10 @@ export class PaymentSubmitDirective {
         @Inject(PaymentRequestService) paymentRequest: PaymentRequestService,
         @Inject(ElementRef) {nativeElement}: ElementRef,
         @Inject(PAYMENT_METHODS) methods: PaymentMethodData[],
-        @Inject(PAYMENT_OPTIONS) options: PaymentOptions,
     ) {
         const requests$ = fromEvent(nativeElement, 'click').pipe(
             switchMap(() =>
-                from(paymentRequest.request({...paymentHost}, methods, options)).pipe(
+                from(paymentRequest.request({...paymentHost}, methods)).pipe(
```
[the remark this answers is not in the document]
Comment (splincode) [reply libs/payment-request/src/directives/payment-submit/payment-submit.directive.ts]: fixed

File: libs/resize-observer/src/directives/resize-observer.directive.ts [hunk 4]
```diff
@@ -37,7 +12,15 @@ export function boxFactory(
         {
             provide: RESIZE_OPTION_BOX,
             deps: [ElementRef],
-            useFactory: boxExtractor,
+            useFactory: ({
+                nativeElement,
+            }: ElementRef<Element>): ResizeObserverBoxOptions => {
+                const attribute = nativeElement.getAttribute(
```
[the remark this answers is not in the document]
Comment (splincode) [reply libs/resize-observer/src/directives/resize-observer.directive.ts:18]: fixed

File: libs/midi/src/utils/input-by-id.ts [hunk 5]
```diff
@@ -21,18 +21,12 @@ export function inputById(id: string): Provider[] {
         {
             provide: MIDI_INPUT,
             deps: [MIDI_ACCESS, MIDI_INPUT_QUERY],
-            useFactory: inputByIdFactory,
+            useFactory: async (
+                midiAccess: Promise<MIDIAccess>,
+                id: string,
+            ): Promise<MIDIInput | undefined> => {
+                return midiAccess.then(access => access.inputs.get(id));
```
[the remark this answers is not in the document]
Comment (splincode) [reply libs/midi/src/utils/input-by-id.ts:28]: fixed

File: libs/canvas/src/contexts/canvas-2d.ts [hunk 6]
```diff
@@ -3,35 +3,30 @@ import {CanvasMethod} from '../interfaces/canvas-method';
...
+                const context = nativeElement.getContext('2d', {
+                    alpha: opaque === null,
+                    desynchronized: desynchronized !== null,
+                });
+
+                if (!context) {
+                    throw new Error('Context of different type was already requested');
+                }
+
+                return context as CanvasRenderingContext2D;
```
[the remark this answers is not in the document]
Comment (splincode) [reply libs/canvas/src/contexts/canvas-2d.ts]: fixed

File: libs/canvas/src/contexts/canvas-2d.ts [hunk 7]
```diff
@@ -3,35 +3,30 @@ import {CanvasMethod} from '../interfaces/canvas-method';
...
-            deps: [
-                ElementRef,
-                // [new Attribute('waOpaque')],
-                // [new Attribute('waDesynchronized')],
-            ],
-            useFactory: canvasContextFactory,
+            deps: [ElementRef],
+            useFactory: (
+                {nativeElement}: ElementRef<HTMLCanvasElement>,
+                opaque: string | null = nativeElement.getAttribute('waOpaque'),
```
[the remark this answers is not in the document]
Comment (splincode) [reply libs/canvas/src/contexts/canvas-2d.ts:19]: fixed

**Review by waterplea — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**